enhancements/cluster-scope-secret-volumes/shared-resource-validation.md

```diff
@@ -0,0 +1,413 @@
+assets managed by the CSO managed Shared Resource CSI Driver Operator, with the deployment of the webhook getting 
+rolled out as different versions of the CSO and Share Resource CSI Driver Operator are applied.
+
+
+### Version Skew Strategy
+
+The validation admission controls discussed here are pattern matching only, stateless checks across calls to it, so the belief is
+upgrades/downgrades in version should roll out the associated Deployments should not affect this enhancement, and the webhooks behavior 
+and what is allowed is thus gated by the version it is at.  Similarly, changes in the Webhook's configuration
+should get applied by existing OCP CVO/CSO processing.  The only API for engaging the webhooks is the k8s webhook APIs themselves.
+The one time read of the ConfigMap allow list for Shared Resources whose name starts with "openshift-*" is impacted
+in the sense that which Shared Resources the webhook allows could change.  However, the removal of any 4.N+1 Shared Resources
+pre-populated by another OCP controller/operator should be handled by the CVO's management of that operator's manifest
+(i.e. other OCP operators that want to provide pre-populated Shared Resources should do so in their manifest.)
+
+And with duplicating checks in the controller relist or CSI Driver interface with the Kubelet, the desired validations
+will be made even if for some reason and upgrade/downgrade of the webhooks are not started yet, or if the new webhooks are not
+functional for some reason. The only difference is how the prevention of the mount is surfaced, as articulated in the goals of the EP.
+
+### Operational Aspects of API Extensions
+
+Carry over from 4.10 (looking to confirm this with 4.10 metrics): expect a low number (single digits) of SharedSecrets
+and SharedConfigMaps.  Expect consumption of those Shares across many Pods etc. across many namespaces.
+
+Also carry over: RBAC objects that grant permission for Pods' ServiceAccounts in those namespaces to use SharedSecrets / SharedConfigMaps 
+in each of those namespaces are needed.  RBAC so that specific SharedSecrets / SharedConfigMaps can be discovered are also
+most likely.
+
+So the 4.10 impact for Shared Resources on the API Server storage i.e. 'etcd' was deemed "no problem".
+
+#### New SLIs for the new admission validations to help cluster administrators and support
+
+Separate metrics for
+= SharedSecret creation/modification failures because of an openshift-* name.  Metric labels for share name, secret name, secret namespace
+- SharedConfigMap creation/modification failures because of an openshift-* name.  Metric labels for share name, configmap name, configmap namespace
+- SharedConfigMap/SharedSecret having the validity condition go to invalid.  Metric labels for share name, reference name and namespace
+- Pod specable creation/modification failure for Shared Resource CSI Volumes without 'readOnly: true'.  Metric labels for namespace, Group/Kind/Version, share name.
```

Comment:
We absolutely need a response time histogram for the webhook if we are reacting all pods, possibly with separate labels for accepted/rejected. If SD team gets a report that it is taking a long time for pods to be created, we need to see to rule out that this webhook was the cause. SLO would need to be something like "99% of all requests return a response within 1 second."

Reply:
ok cool thanks for the confirmation

when we do get down to implementation, we should work with metrics team wrt a new metric vs. the generic webhook one I found